pkg/pillar/cmd/zedagent/attesttask.go

```diff
@@ -209,7 +212,12 @@ func (server *VerifierImpl) SendAttestQuote(ctx *zattest.Context) error {
 		log.Errorf("[ATTEST] Invalid response code")
 		return zattest.ErrControllerReqFailed
 	case attest.ZAttestResponseCode_Z_ATTEST_RESPONSE_CODE_SUCCESS:
-		//XXX Retrieve Storage keys, and integrity token
+		//Retrieve Storage keys, and integrity token
+		iToken := quoteResp.GetIntegrityToken()
+		if iToken == nil {
+			log.Fatalf("[ATTEST] Empty integrity token received")
```

Comment:
Shouldn't have log.Fatal due to something missing or incorrect in a protobuf message we receive from the controller. log.Error plus ignoring the message should be fine.
(I assume the controller will accept and store log messages even if the attestation fails so we can observe any such issues in the logs.)

Reply:
re-factored this a bit, please check now. 
a) no Fatal on receiving empty string or nil
b) in storeIntegrityToken(), we log a warning message for empty/nil token, but we proceed with storing it in the file anyway (to clear out any previous token value )
c) while reading from the file, if read is successful, the content of the file is sent as integrity token. If there is a read error, we don't populate integrity token in the config request.